### PR TITLE
ERAlchemy: Add support for psycopg2

### PR DESCRIPTION
**Pull request by Alexis-benoist**
ERAlchemy wasn't working with PostgreSQL, so I added the support.
Is it a good thing to make it optional? 

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

_You can erase any parts of this template not applicable to your Pull Request._

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

**Comment by dunn** [comment]
> Is it a good thing to make it optional? 

Yes, or at least recommended.  Optional if most people won't want/need it.

**Comment by mikemcquaid** [comment]
👍 to optional.

**Comment by Alexis-benoist** [comment]
Ok, to so I just change to:
depends_on "postgresql"  => :optional

How do I know that the flag is activated? I don't want to install psycopg2 (PG DB API), if I don't have PG.

**Comment by mikemcquaid** [comment]
Yep. It'll install it on `--with-postgresql` in that case.

**Comment by Alexis-benoist** [comment]
I tried to make it optional [here](https://github.com/Alexis-benoist/formula-eralchemy/blob/master/eralchemy.rb).

Though, I cannot see an optional flag doing `brew info eralchemy` (after the copy in the Formula folder). 

Am I doing something wrong?

**Comment by dunn** [comment]
You should be able to just do
```ruby
depends_on :postgresql => :optional
```
Which will generate the option automatically without adding the `option "with-postgresql"` line.

And using `:postgresql` instead of `"postgresql"` lets non-Homebrew psql satisfy the dependency.

**Comment by Alexis-benoist** [comment]
Ok, I made postgreSQL optional, but even with the the flag --with-postgresql, it doesn't install psycopg2.

If it's a non homebrew postgres, will homebrew add to the environment variable the postgres prefix?

**Comment by dunn** [comment]
You need to quote "psycopg2" in `resource("psycopg2").stage do`.

While we're at it, the formula needs OpenSSL.  This should get things working:
```diff
 Formula/eralchemy.rb | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Formula/eralchemy.rb b/Formula/eralchemy.rb
index cf2a61a..aa6a290 100644
--- a/Formula/eralchemy.rb
+++ b/Formula/eralchemy.rb
@@ -12,8 +12,9 @@ class Eralchemy < Formula
   end
 
   depends_on :python if MacOS.version <= :snow_leopard
-  depends_on "graphviz"
   depends_on "pkg-config" => :build
+  depends_on "graphviz"
+  depends_on "openssl"
   depends_on :postgresql => :optional
 
   resource "pygraphviz" do
@@ -45,7 +46,7 @@ class Eralchemy < Formula
     end
 
     if build.with?("postgresql")
-resource(psycopg2).stage do
+      resource("psycopg2").stage do
         system "python", *Language::Python.setup_install_args(libexec/"vendor")
       end
     end
 ```

**Comment by MareoRaft** [comment]
Hey guys.  Can I install psycopg2 onto my computer using brew?  I went on the [psycopg2](http://pythonhosted.org/psycopg2/install.html#install-from-a-package) page and it suggested using fink.  Sorry to bother.

**Comment by Alexis-benoist** [comment]
Great responses guys :)

I will adapt the changes. 

To install psycopg2, Posgres need to be install, and the pg_config program needs to be in PATH env variable. Then you can do `pip install psycopg2`.  [Paraphrasing stackoverflow](http://stackoverflow.com/a/35817509/2422466)

**Comment by dunn** [comment]
@Alexis-benoist otherwise it will link against the old and insecure system OpenSSL:
```
==> Finishing up
ln -s ../Cellar/eralchemy/1.0.1/bin/eralchemy eralchemy
Warning: object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/eralchemy/1.0.1/libexec/vendor/lib/python2.7/site-packages/psycopg2/_psycopg.so

**Comment by Alexis-benoist** [comment]
openssl is installed on my computer so I didn't noticed.

**Comment by Alexis-benoist** [comment]
The code looks good to me, do I make them in only one commit will it be merged?

File: Library/Formula/eralchemy.rb [hunk 1]
```diff
@@ -10,10 +10,11 @@ class Eralchemy < Formula
     sha256 "0d84f7a2fdf7985630487934b280205a603ec591a38c0068921b0be4a1ccc0a2" => :yosemite
     sha256 "9fe62f2286c31683490a44275f99cc92f79ce449b58099f48b1801bd2b9c8e7e" => :mavericks
   end
-
+
```
Comment (dunn) [inline Library/Formula/eralchemy.rb]: Strip any trailing whitespace or `brew audit --strict` will complain.

**Comment by dunn** [comment]
Yeah, squash down to one commit, thanks!

**Comment by Alexis-benoist** [comment]
I squashed the commits :)

**Comment by dunn** [comment]
Crap, I forgot that since everyone should get the OpenSSL update we should add `revision 1` under `sha256`.

And it'd be idea to have the psycopg2 change in one commit, and the OpenSSL/revision in the next.  Sorry for not catching that earlier!

**Comment by Alexis-benoist** [comment]
So you mean adding it after line 11?

2016-04-03 23:12 GMT+02:00 Alex Dunn <notifications@github.com>:

File: Library/Formula/eralchemy.rb [hunk 2]
```diff
@@ -9,11 +9,14 @@ class Eralchemy < Formula
     sha256 "53e20a26f69a3530b499e540a9e33eeb4cf4a125625c0ff9369204c540192eff" => :el_capitan
     sha256 "0d84f7a2fdf7985630487934b280205a603ec591a38c0068921b0be4a1ccc0a2" => :yosemite
     sha256 "9fe62f2286c31683490a44275f99cc92f79ce449b58099f48b1801bd2b9c8e7e" => :mavericks
+    revision 1
```
Comment (mikemcquaid) [inline Library/Formula/eralchemy.rb]: Move this below the `end`, thanks!

**Comment by dunn** [comment]
@Alexis-benoist sorry, I meant after line 5!

File: Library/Formula/eralchemy.rb (hunk 2 above)
Comment (Alexis-benoist) [inline Library/Formula/eralchemy.rb]: I will do it!

**Comment by Alexis-benoist** [comment]
Is it good guys?

**Comment by dunn** [comment]
Thank you! Merged as ac83ad2f5cb4affd7302afe121c9789fda6d7a73 and e0dd2910fb89fe908f421e7e0ef9db92a0418756.

**Pull request closed without merging**